unittests/savanna_misc_tests.cpp

```diff
@@ -142,11 +144,112 @@ BOOST_FIXTURE_TEST_CASE(weak_masking_issue, savanna_cluster::cluster_t) try {
    print("b6", b6);
    BOOST_REQUIRE_EQUAL(qc_s(qc(b6)), qc_s(b5, true)); // b6 should include a strong qc on b5
 
-   BOOST_REQUIRE_EQUAL(A.last_vote, vote_t(b6, true));
-   BOOST_REQUIRE_EQUAL(B.last_vote, vote_t(b6, true));
+   BOOST_REQUIRE_EQUAL(A.last_vote(), vote_t(b6, true));
+   BOOST_REQUIRE_EQUAL(B.last_vote(), vote_t(b6, true));
 
    BOOST_REQUIRE_EQUAL(A.lib_number, b4->block_num());
 
 } FC_LOG_AND_RETHROW()
 
+
+// -----------------------------------------------------------------------------------------------------
+// see https://github.com/AntelopeIO/spring/issues/621 explaining the issue that this test demonstrates.
+//
+// The fix in https://github.com/AntelopeIO/spring/issues/534 for the weak masking issue respected a more
+// conservative version of rule 2. This solved the safety concerns due to the weak masking issue, but it
+// was unnecessarily restrictive with respect to liveness.
+//
+// As a consequence if this liveness issue, finalizers may be stuck voting weak if the QC is not formed
+// quickly enough.
+//
+// This testcase fails prior to https://github.com/AntelopeIO/spring/issues/621 being fixed.
+// -----------------------------------------------------------------------------------------------------
+BOOST_FIXTURE_TEST_CASE(gh_534_liveness_issue, savanna_cluster::cluster_t) try {
+#if 0
```

Comment:
I think we can remove this `#if` and target branch `GH-621-liveness`

Reply:
Right now my gh_621 if based off my gh_544. 
When PR #622 is merged to 1.0, and you have pulled `1.0` into `GH-621-liveness`, then I can change the target.